chips/cc26x0/src/uart.rs

```diff
@@ -199,21 +355,51 @@ impl kernel::hil::uart::UART for UART {
     fn init(&self, params: kernel::hil::uart::UARTParams) {
         self.power_and_clock();
         self.disable_interrupts();
-        self.configure(params);
+        self.configure(params); 
     }
 
     fn transmit(&self, tx_data: &'static mut [u8], tx_len: usize) {
-        if tx_len == 0 { return; }
+        let truncated_len = min(tx_data.len(), tx_len);
+        
+        if truncated_len == 0 {
+            return;
+        }
+        //disable the interrupts until we're ready
+        //self.disable_interrupts();
+
+        //initialize the UART with the passed settings, and then transfer them
+        //to the DMA controller
+        //is there a particular reason we use tx_len rather than truncated_len?
+        //(as with the NRF52, etc.)
```

Comment:
`tx_data.len()` will return the allocated size of the slice (e.g. [0; 32] will return 32), which is why we use `tx_len` as we don't want to transmit the entire buffer every time :) 

assuming `tx_len` is correctly set then `tx_len <= tx_data.len()` should hold